strongbox-web-core/src/main/java/org/carlspring/strongbox/utils/ArtifactControllerHelper.java

```diff
@@ -130,26 +134,30 @@ public static boolean isRangedRequest(HttpHeaders headers)
         }
     }
 
-    public static void setHeadersForChecksums(ArtifactInputStream ais,
+    public static void setHeadersForChecksums(InputStream is,
                                               HttpServletResponse response)
     {
-        ais.getHexDigests()
-           .forEach((k, v) -> response.setHeader(String.format("Checksum-%s",
-                                                               k.toUpperCase().replaceAll("-", "")),
-                                                 v));
+        ArtifactInputStream ais = StreamUtils.findSource(ArtifactInputStream.class, is);
+        ais.getHexDigests().forEach((k,
+                                     v) -> response.setHeader(String.format("Checksum-%s",
+                                                                            k.toUpperCase().replaceAll("-", "")),
+                                                              v));
 
         /*
```

Comment:
@fuss86 , @fuss86 ,

Is this something that could be safely removed, or is it something that could be useful for future debugging?